### PR TITLE
Better stream handling for forked processes

### PR DESCRIPTION
**Pull request by borkweb**
This improves the realtime stream output of forked processes. The behavior can be slightly unpredictable, which I _believe_ is due to how my system is handling the forked processes...but I could be misguided.  The results vary, but interwoven output is the more common result.

:movie_camera: http://p.tri.be/2aBSUW

### CODE REVIEW

**Comment by lucatume** [comment]
@borkweb I've pushed an iteration on this code to fall-back on serial execution when the host OS does not support the Process Control extension.
To my knowledge that would be only Windows (WSL does support it though); so that fallback will cover all the bases; please review.

File: src/process.php [hunk 1]
```diff
@@ -185,21 +210,39 @@ function check_status_or( callable $process, callable $else = null ) {
 function parallel_process( $items, $command_process ) {
 	$process_children = [];
 
-	foreach ( $items as $item ) {
-		$pid = pcntl_fork();
-		if ( $pid === -1 ) {
-			echo magenta( "Unable to fork processes.\n" );
-			exit( 1 );
+	if (function_exists('pcntl_fork')) {
```
Comment (borkweb) [inline src/process.php:213]:
```suggestion
	if ( function_exists( 'pcntl_fork' ) ) {
```

File: src/process.php [hunk 2]
```diff
@@ -185,21 +210,39 @@ function check_status_or( callable $process, callable $else = null ) {
...
+		return get_status_of_forked_children( $process_children );
+	}
+
+	/*
+	 * If Process Control functions are not available or are disabled, then we execute the commands serially.
+	 * Nothing "parallel" here.
+	 */
+	foreach ( $items as $item ) {
+		$status = $command_process( $item );
+		if($status !== 0){
```
Comment (borkweb) [inline src/process.php:238]:
```suggestion
		if ( $status !== 0 ) {
```

**Comment by borkweb** [comment]
Excellent changes, @lucatume!

**Pull request merged**